File: guides/micronaut-data-mongodb-asynchronous/java/src/main/java/example/micronaut/FruitService.java
package example.micronaut;

import io.micronaut.core.annotation.NonNull;
import org.reactivestreams.Publisher;

import java.util.List;

interface FruitService {

    Publisher<Fruit> list();

    Publisher<Fruit> save(Fruit fruit);

    Publisher<Fruit> find(@NonNull String id);

    Publisher<Fruit> findByNameInList(List<String> name);
}

File: guides/micronaut-data-mongodb-asynchronous/python/src/example/micronaut/fruit_service.py
from typing import List

from org.reactivestreams import Publisher

from .fruit import Fruit


class FruitService:
    def list(self) -> Publisher[Fruit]: ...

    def save(self, fruit: Fruit) -> Publisher[Fruit]: ...

    def find(self, id: str) -> Publisher[Fruit]: ...

    def findByNameInList(self, names: List[str]) -> Publisher[Fruit]: ...
